pkg/controllers/resourcechange/resourcechange_controller_test.go

```diff
@@ -1757,131 +1855,546 @@ func TestTriggerAffectedPlacementsForUpdatedClusterRes(t *testing.T) {
+					Version:   "v1",
+					Kind:      "Deployment",
+					Name:      "test-deployment-no-match",
+					Namespace: "test-namespace",
+				},
+			},
+			resource: func() *unstructured.Unstructured {
+				testDeployment := &appv1.Deployment{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "Deployment",
+						APIVersion: "apps/v1",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-deployment-no-match",
+						Namespace: "test-namespace",
+						Labels: map[string]string{
+							"app": "no-match",
+						},
+					},
+				}
+				return createDeploymentUnstructured(testDeployment)
+			}(),
+			informerManager: func() informer.Manager {
+				// Create a ResourcePlacement that won't match (different labels)
+				nonMatchingRP := &placementv1beta1.ResourcePlacement{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "non-matching-rp",
+						Namespace: "test-namespace",
+					},
+					Spec: placementv1beta1.PlacementSpec{
+						ResourceSelectors: []placementv1beta1.ClusterResourceSelector{
+							{
+								Group:   "apps",
+								Version: "v1",
+								Kind:    "Deployment",
+								LabelSelector: &metav1.LabelSelector{
+									MatchLabels: map[string]string{
+										"app": "different-app",
+									},
+								},
+							},
+						},
+					},
+				}
+				uMap, _ := runtime.DefaultUnstructuredConverter.ToUnstructured(nonMatchingRP)
+				rpObjects := []runtime.Object{&unstructured.Unstructured{Object: uMap}}
+
+				return &fakeInformerManager{
+					listers: map[schema.GroupVersionResource]*fakeLister{
+						utils.ClusterResourcePlacementGVR: {
+							objects: []runtime.Object{},
+						},
+						utils.ResourcePlacementGVR: {
+							objects: rpObjects,
+						},
+					},
+				}
+			}(),
+			wantPlacementControllerEnqueued:         []string{},
+			wantResourcePlacementControllerEnqueued: []string{},
+		},
+		"cluster-scoped resource with empty informer data": {
+			key: keys.ClusterWideKey{
+				ResourceIdentifier: fleetv1alpha1.ResourceIdentifier{
+					Group:   "",
+					Version: "v1",
+					Kind:    "Namespace",
+					Name:    "other-namespace",
+				},
+			},
+			resource: func() *unstructured.Unstructured {
+				otherResource := &corev1.Namespace{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "Namespace",
+						APIVersion: "v1",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "other-namespace",
+						Labels: map[string]string{
+							"app": "other",
+						},
+					},
+				}
+				return createNamespaceUnstructured(otherResource)
+			}(),
+			informerManager: &fakeInformerManager{
+				listers: map[schema.GroupVersionResource]*fakeLister{
+					utils.ClusterResourcePlacementGVR: {
+						objects: []runtime.Object{},
+					},
+					utils.ResourcePlacementGVR: {
+						objects: []runtime.Object{},
+					},
+				},
+			},
+			wantPlacementControllerEnqueued:         []string{},
+			wantResourcePlacementControllerEnqueued: []string{},
+		},
+		"cluster-scoped resource with multiple CRPs": {
```

Comment:
ditto, add tests for mixing CRP and RP in the lister?

Reply:
the first case is mixing both in the lister but it should not matter as it will only trigger one controller.